autogtp/Game.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+    This file is part of Leela Zero.
+    Copyright (C) 2017 Marco Calignano
+
+    Leela Zero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Leela Zero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include<QUuid>
+#include "Game.h"
+
+Game::Game(QString weights, QTextStream *out) :
```

Comment:
Can out be a reference instead of a pointer? That would make the code a bit cleaner.

Reply:
Reference does not work because the copy constructor is private and I cannot assign it to a member variable.

Reply:
Does it not work if you put it in the constructor's base initializer list?

i.e. 

Game::Game(QString weights, QTextStream& out) 
 : m_out(out)
 {
....
}

Maybe I remember wrong, but I think this avoids the problem.